demo

```diff
@@ -55,16 +68,118 @@ if [[ "$?" -eq 1 ]]; then
+#       The following code will only be run in the case ./demo up [args]
+#
+# ------------------------------------------------------------------------------
+
+echo "Starting the Mender demo environment..."
+
+docker-compose \
+    -f docker-compose.yml \
+    -f docker-compose.storage.minio.yml \
+    -f docker-compose.demo.yml \
+    $client \
+    "$@" -d
+
+# DC_PID=$!
+
+USER='myusername@example.com'; PASSWORD='mysecretpassword'
+MENDER_SERVER_USER="${USER}:${PASSWORD}"
+MENDER_SERVER_URI='https://localhost'
+
+# Block until the useradm service returns an HTTP response.
+echo "Waiting for the useradm service to start up"
+curl --silent -k -X POST -u $MENDER_SERVER_USER\
+     --fail\
+     --retry-connrefused\
+     --retry 60\
+     --retry-delay 10\
+     --retry-max-time 120\
+     $MENDER_SERVER_URI/api/management/v1/useradm/auth/login > /dev/null
+
+retval=$?
+case $retval in
+    22) ;; # Server 400 error, ie, server is up - continue.
+    0)  ;; # User exists - continue.
+    *)
+       echo "Failed to access the mender-useradm service within the allotted two minutes."
+       echo "Please check the docker-compose setup."
+       echo "exit code" $retval
+       exit $retval
+       ;;
+esac
+
+echo "Creating a new user..."
+
+# Create a new user.
+docker exec integration_mender-useradm_1 /usr/bin/useradm create-user --username=${USER} --password=${PASSWORD}
+
+retval=$?
+case $retval in
+     0) echo "New user created." ;;
+     5) ;; # User already exists.
+     *) exit $retval ;;
+esac
+
+echo "Getting JWT Token"
+
+JWT=$(curl --silent -k -X POST -u $MENDER_SERVER_USER\
+           --fail\
+           --retry-connrefused\
+           --retry 60\
+           --retry-delay 10\
+           --retry-max-time 120\
+           $MENDER_SERVER_URI/api/management/v1/useradm/auth/login)
+
+retval=$?
+if [ $retval -ne 0 ]; then
+    echo "Failed to get the 'JWT' token from the useradm service."
+    echo "This is needed in order to use the demo Artifact."
+    exit $retval
+fi
+
+echo "Uploading the artifact to the server..."
+errout=$(curl -k --request POST\
+              --silent\
+              --show-error\
+              --retry-connrefused\
+              --retry-delay 10\
+              --retry 60 \
+              --retry-max-time 120\
+              --header "Authorization: Bearer ${JWT}"\
+              --form "size=${ARTIFACT_SIZE_BYTES}"\
+              --form "artifact=@${DEMO_ARTIFACT_NAME}"\
+              "${MENDER_SERVER_URI}/api/management/v1/deployments/artifacts" | jq .error)
+
+retval=$?
+
+case "$errout" in
+   " Artifact not unique"*) echo "Artifact already exists on the server" ;; # Artifact is already on the server.
+    "") echo "Artifact uploaded to the demo server" ;;
+    *) echo "Uploading the demo artifact failed with error: " $errout
+       exit $retval ;;
+esac
+
+trap "docker-compose \
+    -f docker-compose.yml \
+    -f docker-compose.storage.minio.yml \
+    -f docker-compose.demo.yml \
+    down" SIGINT
```

Comment:
Btw, this should be moved up to right below the `up`, so that it works no matter where in the execution you press Ctrl-C.